test/parallel/test-vm-codegen.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
```

Comment:
This also needs tests for unexpected types of `contextCodeGeneration` and its properties.